api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -89,12 +93,12 @@ public R or(R leftResult, R rightResult) {
       return null;
     }
 
-    public <T> R in(BoundReference<T> ref, Literal<T> lit) {
-      return null;
+    public <T> R in(BoundReference<T> ref, LiteralSet<T> literalSet) {
```

Comment:
I don't think this needs to leak the `LiteralSet` implementation class through the public API. That should be a package-private class and this can use `Set<T>` instead.

Reply:
Yep, thanks for the comments. Updated them accordingly.

Reply:
As only `BoundSetPredicate` can access `LiteralSet`, I will make `LiteralSet` as `BoundSetPredicate`'s private inner class.